pipeline/inferrer/feature_training/src/aws.py

```diff
@@ -1,28 +1,39 @@
 import boto3
+import os
+import requests
 from botocore.exceptions import ClientError
+from .logging import get_logger
+
+logger = get_logger(__name__)
+
+
+def get_ecs_container_metadata():
+    endpoint = os.environ["ECS_CONTAINER_METADATA_URI_V4"]
+    req = requests.get(endpoint)
+    return req.json()
 
 
 def get_s3_client():
     try:
         s3_client = boto3.client("s3")
     except ClientError as e:
-        print(f"Failed to create s3 client: {e}")
+        logger.info(f"Failed to create s3 client: {e}")
         raise e
     return s3_client
 
 
 def put_object_to_s3(binary_object, key, bucket_name):
     s3_client = get_s3_client()
-    print("Uploading object to S3...")
+    logger.info("Uploading object to S3...")
     s3_client.put_object(Bucket=bucket_name, Key=key, Body=binary_object)
-    print("Uploaded object to S3.")
+    logger.info("Uploaded object to S3.")
 
 
 def put_ssm_parameter(path, value, description):
     ssm_client = boto3.client("ssm")
 
-    print(f"Updating SSM path `{path}` to `{value}`...")
+    logger.info(f"Updating SSM path `{path}` to `{value}`...")
```

Comment:
Minor: Python best practice is not to do string interpolation inside logging calls, but to pass the parameters as separate arguments and let them be interpolated lazily, i.e.
```suggestion
    logger.info("Updating SSM path %r to %r", path, value)
```

Python will only actually perform the string interpolation if this message is going to be sent anywhere; if it’s below the current log level you save an interpolation.

(Which granted isn't a huge benefit here, but it's a good habit to get into.)